### PR TITLE
detect stdc++ automatically

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

File: librocksdb_sys/build.sh [hunk 1]
```diff
@@ -96,12 +111,24 @@ function compile_rocksdb() {
...
+function find_stdcxx() {
+    if g++ --version &>/dev/null; then
+        CXX=g++
+    elif clang++ --version &>/dev/null; then
+        CXX=clang++
+    else
+        error failed to find valid cxx compiler.
+    fi
+
+    $CXX --print-file-name libstdc++.a
```
Comment (siddontang) [inline librocksdb_sys/build.sh]: this can't work ok in mac os x directly. 
e,g, if you use brew install g++ 4.9, the g++ here you call is still g++ 4.2.

Comment (siddontang) [inline librocksdb_sys/build.sh]: If we don't install static c++, these will output `libstdc++.a`

Comment (BusyJay) [inline librocksdb_sys/build.sh]: OS X doesn't provide a static libstdc++ actually.

Output libstdc++.a is ok. `build.rs` will check if the output is absolute.

Comment (siddontang) [inline librocksdb_sys/build.sh]: No, in my mac, I can use `g++4.9 --print-file-name libstdc++.a` to get this. but I think we don't need to consider Mac OS X crossing machines.

**Comment by siddontang** [comment]
LGTM

File: librocksdb_sys/build.sh (hunk 1 above)
Comment (BusyJay) [inline librocksdb_sys/build.sh]: If `g++4.9 --print-file-name libstdc++.a` can print the path, then L120 should handle this properly.

Comment (BusyJay) [inline librocksdb_sys/build.sh]: Oh, I see what you mean now. I don't think we need to handle this situation. If g++4.9 is not the default c++ compiler, user can pass the stdc++ library explicitly by ROCKSDB_OTHER_STATIC. Handle all possible situation is complicated and unnecessary.

**Comment by BusyJay** [comment]
PTAL

**Comment by c4pt0r** [comment]
LGTM

**Pull request merged**